composables/use-subscription.ts

```diff
@@ -57,14 +57,33 @@ export function useSubscription() {
     props: getPaywallModalProps(),
   })
 
-  async function openPaywallModal(props: PaywallModalProps = {}) {
+  async function openPaywallModal(props: (PaywallModalProps & {
+    utmCampaign?: string
+    utmMedium?: string
+    utmSource?: string
+  }) = {}) {
     if (paywallModal.isOpen) {
       paywallModal.close()
     }
 
+    const utmOverrides: {
+      utmCampaign?: string
+      utmMedium?: string
+      utmSource?: string
+    } = {}
+    if (props.utmCampaign) {
+      utmOverrides.utmCampaign = props.utmCampaign
+    }
+    if (props.utmMedium) {
+      utmOverrides.utmMedium = props.utmMedium
+    }
+    if (props.utmSource) {
+      utmOverrides.utmSource = props.utmSource
+    }
     return paywallModal.open({
       ...getPaywallModalProps(),
       ...props,
+      onSubscribe: () => startSubscription(utmOverrides),
```

Comment:
if u use utm from props, no need to add this
https://github.com/likecoin/3ook-com/blob/34e057abb0dfa7061244928e69bdd64dcf8d4b08/composables/use-subscription.ts#L51